### PR TITLE
[TEST] Add test about batch processor with current_thread async runtime

### PR DESCRIPTION
**Pull request by ThomsonTan**
## Changes

Add test to reproduce the deadlock during shutdown of batch processor when configured with single thread async runtime. Related to issue https://github.com/open-telemetry/opentelemetry-rust/issues/2066.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/logs/log_processor.rs [hunk 1]
```diff
@@ -812,6 +812,26 @@ mod tests {
...
+
+    #[tokio::test(flavor = "current_thread")]
+    async fn test_batch_log_processor_shutdown_with_async_runtime() {
+        let exporter = InMemoryLogsExporterBuilder::default()
+            .keep_records_on_shutdown()
+            .build();
+        let processor = BatchLogProcessor::new(
+            Box::new(exporter.clone()),
+            BatchConfig::default(),
+            runtime::Tokio,
```
Comment (TommyCpp) [inline opentelemetry-sdk/src/logs/log_processor.rs:824]: With `Tokio` this probably will cause a deadlock. By design this should work with `TokioCurrentThread`

**Comment by ThomsonTan** [comment]
Reproduced the hang in below iteration, ignore running the test in the latest iteration.

https://github.com/open-telemetry/opentelemetry-rust/actions/runs/10907111063/job/30269977617?pr=2123

File: opentelemetry-sdk/src/logs/log_processor.rs [hunk 2]
```diff
@@ -812,6 +812,24 @@ mod tests {
         assert_eq!(1, exporter.get_emitted_logs().unwrap().len())
     }
 
+    #[tokio::test(flavor = "current_thread")]
+    #[ignore = "Enable when this is fixed in the batch processor"]
```
Comment (lalitb) [inline opentelemetry-sdk/src/logs/log_processor.rs:816]: nit - add link to the issue.

File: opentelemetry-sdk/src/logs/log_processor.rs (hunk 1 above)
Comment (lalitb) [reply opentelemetry-sdk/src/logs/log_processor.rs:824]: @ThomsonTan Can we add a test for these combinations - with/without `current_thread` annotation, and using `runtime::TikioCurrentThread` ?

Comment (ThomsonTan) [reply opentelemetry-sdk/src/logs/log_processor.rs:824]: Added 3 more tests to cover the combination of flavor and our runtime parameter.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**